identity-service/src/routes/trackListens.js

```diff
@@ -436,4 +437,48 @@ module.exports = function (app) {
       listenMap
     })
   }))
+
+  /**
+   * Gets user listen records in bulk.
+   * 
+   * GET query parameters:
+   *  startTime (string) the start time to fetch listens from (in unix seconds format)
+   *  limit (number) the number of records to fetch from each user listens and anonymous listens
+   */
+  app.get('/listens/bulk', handleResponse(async (req, res) => {
+    let { startTime, limit } = req.query
+    if (!startTime || !limit) {
+      return errorResponseBadRequest('Please provide a startTime and limit')
+    }
+
+    limit = Math.min(limit, 1000)
+    let createdAtMoment
+    try {
+      createdAtMoment = moment.unix(startTime)
+      if (!createdAtMoment.isValid()) throw new Error()
+    } catch (e) {
+      return errorResponseBadRequest(`Provided startTime ${startTime} not parseable`)      
+    }
+
+    const userListens = await models.UserTrackListen.findAll({
+      attributes: { exclude: ['id'] },
+      where: {
+        createdAt: { [models.Sequelize.Op.gt]: createdAtMoment.toDate() }
+      },
+      order: [['createdAt', 'ASC'], ['trackId', 'ASC']],
+      limit
+    })
+
+    const anonListens = await models.TrackListenCount.findAll({
+      where: {
+        createdAt: { [models.Sequelize.Op.gt]: createdAtMoment.toDate() }
+      },
+      order: [['createdAt', 'ASC'], ['trackId', 'ASC']],
+      limit
+    })
+
+    return successResponse({
+      listens: [...userListens, ...anonListens]
```

Comment:
Not sure about this, but isn't there extra metadata from the ORM model instance? Maybe it's just dropped though when converting to json

Reply:
what extra metadata are you talking about? we want all the fields here (minus ID that is intentionally excluded)

Reply:
i think joey is talking about the functions and propertiesmodel instances. when you query through sequelize via findOne or findAll you can interact with each object and call functions on it's model and save it's value etc. but pretty sure when it gets serialized it removes all that stuff that used to be on the object

Reply:
ah, yeah those are all removed